### ISSUE TITLE
Adding while-guard to path condition must be under substitution

### ISSUE THREAD

**Issue by cavilacion** [zsusag/symbolic_semantics_of_probabilistic_programs#8]
When symbolically executing a random walk on Z with symbolic inital variable by the following program:
```
fn main(x:Real) {
  while (x>0) {
    u ~ bern(0.5);
    x := x + 2*u-1;
  }
}
```
the conjuncts in the path condition from the while guard are not added under substitution.
For example, when executing the above program with `--max-iterations 1` yields
```
Number of Paths: 3
Path 1:
	Path Condition: ¬(x > 0)
	Probability:  Boole[x<=0]
	Observations:
	Observations Probability:  1
	Sigma: σ(x) = x
	Terminated: No
Path 2:
	Path Condition: x > 0 ∧ ¬(UNIFORM_0 < 1/2) ∧ ¬(x > 0)
	Probability:  0
	Observations:
	Observations Probability:  1
	Sigma: σ(u) = UNIFORM_0, σ(x) = x + 0 - 1
	Terminated: Yes
Path 3:
	Path Condition: x > 0 ∧ UNIFORM_0 < 1/2
	Probability:  1/2*Boole[-x<=0]*Boole[x!=0]
	Observations:
	Observations Probability:  1
	Sigma: σ(u) = UNIFORM_0, σ(x) = x + 2 - 1
	Terminated: Yes
```
But in Path 2 there should be a conjunct `¬(x-1 > 0)` rather than `¬(x > 0)`.
Also, shouldn't there be a conjunct `¬(x+1 > 0)` in Path 3?

**Comment by zsusag** [comment]
Yes, you are correct! I have pushed a change which fixes it: 008a4fc4f13b4cf45c25195f0758ff84d27fdb16

Now when you run your example, you get the following paths:
```
Number of Paths: 3
Number of Removed Paths: 0
Number of Samples: 1
Path 1:
	Path Condition: x > 0 ∧ UNIFORM_0 < 1/2 ∧ (x + 2 - 1) > 0
	Observations: 
	Sigma: σ(u) = UNIFORM_0, σ(x) = x + 2 - 1
	Terminated: Yes
Path 2:
	Path Condition: ¬(x > 0)
	Observations: 
	Sigma: σ(x) = x
	Terminated: No
Path 3:
	Path Condition: x > 0 ∧ ¬(UNIFORM_0 < 1/2) ∧ ¬((x + 0 - 1) > 0)
	Observations: 
	Sigma: σ(u) = UNIFORM_0, σ(x) = x + 0 - 1
	Terminated: Yes
```

I know that there is an issue with simplifying the expressions like `x + 0 - 1`, although it doesn't appear in any of the case studies so I'm not going to solve that bug today... the issue is because it is actually written `(x + 0) - 1` in the AST and my simplification procedure is too simplistic and doesn't handle these cases (i.e., it will only simplify nodes which are all constants).

**Comment by zsusag** [comment]
Whoops, actually that was wrong, here is the correct output from a further bug fix:
```
Number of Paths: 3
Number of Removed Paths: 0
Number of Samples: 1
Path 1:
	Path Condition: x > 0 ∧ ¬(UNIFORM_0 < 1/2) ∧ ¬((x + 0 - 1) > 0)
	Observations: 
	Sigma: σ(u) = UNIFORM_0, σ(x) = x + 0 - 1
	Forced termination: Yes
Path 2:
	Path Condition: ¬(x > 0)
	Observations: 
	Sigma: σ(x) = x
	Forced termination: No
Path 3:
	Path Condition: x > 0 ∧ UNIFORM_0 < 1/2 ∧ ¬((x + 2 - 1) > 0)
	Observations: 
	Sigma: σ(u) = UNIFORM_0, σ(x) = x + 2 - 1
	Forced termination: Yes
```
